CODING_STANDARDS.md

```diff
@@ -360,21 +360,20 @@ For example:
 my_function(thd, db_name);
 ```
 
-### Types
+### Integer types
 
-In general the usage of types such as `char`, `int` and `long` should be discouraged but there are shortened versions of the unsigned variants available for these in `my_global.h`.
-They can be different sizes across platforms and `char` can be either unsigned or signed depending on platform, and therefore are not portable.
-Instead these should be used as appropriate:
+The usage of types `long`  and `unsigned long` (and its `ulong` alias)
+in new code is *strongly* discouraged. Its use brings no advantages,
+only portability problems between Windows and Unixes.
 
-* 8-bit signed / unsigned int -> `int8` / `uint8`
-* 16-bit signed / unsigned int -> `int16` / `uint16`
-* 32-bit signed / unsigned int -> `int32` / `uint32`
-* 64-bit signed / unsigned int -> `int64` / `uint64`
-* Integer file descriptor -> `File`
-* Integer socket descriptor -> `my_socket`
+Instead of using `long`, use `size_t` and `ptrdiff_t` where appropriate,
+buffer sizes for example. For integer socket descriptor use `my_socket`.
 
-`size_t` and `ptrdiff_t` are used in the source where appropriate, buffer sizes for example.
-It should be noted that these are implementation dependent but are useful when used in the correct context.
+You may use types with fixed length, int32_t and similar, too. Yet, on all platforms we currently support,
```

Comment:
~~One day someone might ask if we support embedding [in an Arduino](https://docs.arduino.cc/language-reference/en/variables/data-types/int/).~~

Reply:
Thus , "we currently support". If the need arises, we'll port and cope with "long long" being 2 bytes. *If* the need arieses

Reply:
Also, what's special in an Ardunio? https://learn.sparkfun.com/tutorials/data-types-in-arduino/defining-data-types.
it does not have 64bit shorts, and does not have 16bit ints.

Reply:
As linked in my ~~scratched out~~ comment, it does have 16bit ints.

> **int** (16 bit) - signed number from -32768 to 32767.

Reply:
We won't port on Ardunio 16 bit integers. Those heroes who succeed, can then fix that paragraph - it will the the smallest of their problems. But is much more likely there won't even be another 32bit OS mariadb will run on.